generic/dockerfile/best-practice/use-absolute-workdir.yaml

```diff
@@ -1,12 +1,31 @@
 rules:
   - id: use-absolute-workdir
-    pattern: WORKDIR $WORD
+    pattern-either: 
+      - patterns:
+        - pattern: WORKDIR $VALUE
+        - metavariable-pattern:
+            metavariable: $VALUE
+            patterns:
+              - pattern-not-regex: (\/.*)
+      - patterns:
+        - pattern: 
+            ENV $VAR=$VALUE
+            ...
+            $CMD ${$VAR}
+        - metavariable-pattern:
+            metavariable: $VALUE
+            patterns:
+              - pattern-not-regex: (\/.*)
+        - metavariable-pattern:
+            metavariable: $CMD
+            pattern: WORKDIR
+        - focus-metavariable: $CMD
     message: >-
       Detected a relative WORKDIR. Use absolute paths. This prevents
       issues based on assumptions about the WORKDIR of previous containers.
     severity: WARNING
     languages: [generic]
-    fix: WORKDIR /$WORD
+    #fix: WORKDIR /$VALUE
```

Comment:
is this commented because the fix isn't effective at this time?

Reply:
That is correct, there are cases that do not work, and I can't write an effective fix for now. I think because of limitations of the generic parser that is not enough context-aware. I will take a stab at porting this rule to dockerfile parser at a later time.